src/prototype_creep_resources.js

```diff
@@ -360,7 +360,9 @@ const filterTransferrables = function(creep, object) {
     return false;
   }
 
-  if (object.structureType !== STRUCTURE_STORAGE && object.energy === object.energyCapacity) {
+  if (object.structureType === STRUCTURE_STORAGE ?
+    _.sum(object.store) + _.sum(creep.carry) > object.storeCapacity :
+    object.energy === object.energyCapacity) {
```

Comment:
This whole energy-vs-store needs to be moved out to a helper function. It's used multiple places in the code, and ripe for errors.